// fadab-mysql-helper/lib/databaseTransactions/FadabHelper.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FadabHelper = void 0;
var queryTransaction_1 = require("./queryTransaction");
var selectTransaction_1 = require("./selectTransaction");
var findTransaction_1 = require("./findTransaction");
var insertTransaction_1 = require("./insertTransaction");
var updateTransaction_1 = require("./updateTransaction");
var deleteTransaction_1 = require("./deleteTransaction");
var countTransactions_1 = require("./countTransactions");
var upsertTransaction_1 = require("./upsertTransaction");
var bulkInsertTransaction_1 = require("./bulkInsertTransaction");
var FadabHelper = /** @class */ (function () {
    function FadabHelper() {
        this.baseTable = '';
        this.queryAsync = queryTransaction_1.queryAsync;
    }
    FadabHelper.prototype.selectAsync = function (options) {
        return selectTransaction_1.selectAsync(this.baseTable, options);
    };
    FadabHelper.prototype.findOneAsync = function (where) {
        return findTransaction_1.findOneAsync(this.baseTable, where);
    };
    FadabHelper.prototype.insertAsync = function (values, ignore) {
        return insertTransaction_1.insertAsync(this.baseTable, values, ignore);
    };
    FadabHelper.prototype.updateAsync = function (values, where) {
        return updateTransaction_1.updateAsync(this.baseTable, values, where);
    };
    FadabHelper.prototype.deleteAsync = function (where) {
        return deleteTransaction_1.deleteAsync(this.baseTable, where);
    };
    FadabHelper.prototype.countAsync = function (options) {
        return countTransactions_1.countAsync(this.baseTable, options);
    };
    FadabHelper.prototype.bulkInsertAsync = function (values, ignore) {
        return bulkInsertTransaction_1.bulkInsertAsync(this.baseTable, values, ignore);
    };
    FadabHelper.prototype.upsertAsync = function (values) {
        return upsertTransaction_1.upsertAsync(this.baseTable, values);
    };
    return FadabHelper;
}());
exports.FadabHelper = FadabHelper;
